qiskit/circuit/library/arithmetic/inverse_chebyshev.py

```diff
@@ -0,0 +1,298 @@
+
+
+class InverseChebyshev(BlueprintCircuit):
+    r"""Piecewise Chebyshev approximation of the inverse function.
+
+    Building block of HHL. For a given constant :math:`C` and error tolerance :math:`\epsilon`,
+    this class implements a piecewise polynomial approximation on :math:`n` qubits to
+    :math:`arcsin(C/x)`, where :math:`x \in [a,2^n - 1]` and :math:`a` depends on :math:`\epsilon`.
+
+    The values of the parameters are calculated according to [1].
+
+    Examples:
+        >>> from qiskit import QuantumCircuit
+        >>> from qiskit.circuit.library.arithmetic.inverse_chebyshev import InverseChebyshev
+        >>> num_qubits = 4
+        >>> err_tolerance = 0.1
+        >>> inv_cheb = InverseChebyshev(num_qubits, err_tolerance)
+        >>> inv_cheb._build()
+        >>> qc = QuantumCircuit(num_qubits + 1 + inv_cheb.num_ancillas)
+        >>> qc.h(list(range(num_qubits)))
+        <qiskit.circuit.instructionset.InstructionSet object at 0x00000297722900B8>
+        >>> qc.append(inv_cheb.to_instruction(), list(range(qc.num_qubits)))
+        >>> qc.draw()
+              ┌───┐┌────────────┐
+         q_0: ┤ H ├┤0           ├
+              ├───┤│            │
+         q_1: ┤ H ├┤1           ├
+              ├───┤│            │
+         q_2: ┤ H ├┤2           ├
+              ├───┤│            │
+         q_3: ┤ H ├┤3           ├
+              └───┘│            │
+         q_4: ─────┤4           ├
+                   │            │
+         q_5: ─────┤5           ├
+                   │            │
+         q_6: ─────┤6           ├
+                   │   inv_cheb │
+         q_7: ─────┤7           ├
+                   │            │
+         q_8: ─────┤8           ├
+                   │            │
+         q_9: ─────┤9           ├
+                   │            │
+        q_10: ─────┤10          ├
+                   │            │
+        q_11: ─────┤11          ├
+                   │            │
+        q_12: ─────┤12          ├
+                   │            │
+        q_13: ─────┤13          ├
+                   └────────────┘
+
+    References:
+
+        [1]: Carrera Vazquez, A., Hiptmair, R., & Woerner, S. (2020).
+             Enhancing the Quantum Linear Systems Algorithm using Richardson Extrapolation.
+             `arXiv:2009.04484 <http://arxiv.org/abs/2009.04484>`_
+
+        [2]: Haener, T., Roetteler, M., & Svore, K. M. (2018).
+             Optimizing Quantum Circuits for Arithmetic.
+             `arXiv:1805.12445 <http://arxiv.org/abs/1805.12445>`_
+    """
+
+    def __init__(self,
+                 num_state_qubits: Optional[int] = None,
+                 epsilon: Optional[float] = None,
+                 constant: Optional[float] = None,
+                 kappa: Optional[float] = None,
+                 name: str = 'inv_cheb') -> None:
+        """
+        Args:
+            num_state_qubits: number of qubits representing the state.
+            epsilon: accuracy of the approximation.
```

Comment:
The usual way to do this is 
```python
def __init__(self, ..., epsilon: float = 1e-2, ..)
```
then you don't explicity have to write that it defaults to ``1e-2`` (since its obvious from the signature) and you also don't have to do the `if None` comparisons below 🙂